pkg/cache/v2/cache.go

```diff
@@ -145,30 +210,113 @@ func (r *RawResponse) GetDiscoveryResponse() (*discovery.DiscoveryResponse, erro
 	return marshaledResponse.(*discovery.DiscoveryResponse), nil
 }
 
+// GetDeltaDiscoveryResponse performs the marshalling the first time its called and uses the cached response subsequently.
+// This is necessary because the marshalled response does not change across the calls.
+// This caching behavior is important in high throughput scenarios because grpc marshalling has a cost and it drives the cpu utilization under load.
+func (r *RawDeltaResponse) GetDeltaDiscoveryResponse() (*discovery.DeltaDiscoveryResponse, error) {
+	marshaledResponse := r.marshaledResponse.Load()
+
+	if marshaledResponse == nil {
+		marshaledResources := make([]*discovery.Resource, len(r.Resources))
+
+		for i, resource := range r.Resources {
+
+			marshaledResource, err := MarshalResource(resource)
+			if err != nil {
+				return nil, err
+			}
+			version, err := HashResource(resource)
+			if err != nil {
+				panic(err)
+			}
+
+			name := GetResourceName(resource)
+			marshaledResources[i] = &discovery.Resource{
+				Name:    name,
+				Aliases: []string{name},
```

Comment:
I went with name here due to the information stated within the discovery proto code:
```go
	// The aliases are a list of other names that this resource can go by.
	Aliases []string `protobuf:"bytes,4,rep,name=aliases,proto3" json:"aliases,omitempty"`
```
It seems like the literal meaning is name, I'm not sure what else we could use here? Unless we derived a name from particular information in the cache?

Reply:
The key thing to notice from those docs is that it says "other", it will use these in addition to the Name so there is no reason to duplicate it

An example usage in Envoy looks like 

```
  std::set<std::string> added_resource_ids;
  for (const auto& resource : added_resources) {
    added_resource_ids.emplace(resource.get().name());
    std::copy(resource.get().aliases().begin(), resource.get().aliases().end(),
              std::inserter(added_resource_ids, added_resource_ids.end()));
...
```

which would just dedup it away, so there's no harm behind perhaps some confusion. I think we should just leave this empty.

cc @htuch in case we feel like the docs could be clearer here, maybe not clear that this is completely optional?

Reply:
Gotcha. Yup we can probably pull that out it seems like it's not needed if that is the case. I misunderstood that this was an additional list of potential names + the original name. I'll make the change!

Reply:
@dmitri-d for docs update as this is VHDS specific. We plan on deprecating aliases in the future, but for now they are essentially there for VHDS to fetch multiple virtual hosts in a single resource on-demand.